salt/files/site-backup-to-s3.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+#
+# This file is managed by Dogsbody Technology Ltd.
+#   https://www.dogsbody.com/
+#
+# Description:  Backup multiple directories and upload to AWS S3
+#
+# Usage:  $0
+#
+
+set -euo pipefail
+
+source /home/sysadmin-tools/aws-settings.local
+
+export AWS_ACCESS_KEY_ID
+export AWS_SECRET_ACCESS_KEY
+export AWS_DEFAULT_REGION
+
+if [ -z "$FOLDER2S3BACKUPSRC" ]; then
+    echo "Error: The variables FOLDER2S3BACKUPSRC have not been set"
+    exit 4
+fi
+
+if [ ! -x "${AWS_CLI}" ]; then
+    echo "Error: The aws executable is not installed"
+    exit 7
+fi
+
+for SITE in "${FOLDER2S3BACKUPSRC[@]}"; do
+    SITENAME="${SITE//[^a-zA-Z0-9]/_}"
+    SITENAME="${SITENAME//^___/}"
```

Comment:
What does this line do?

Reply:
This line converts the site name into an alphanumeric (plus underscores) string, so it ensures it is valid for AWS.
> SITENAME="${SITE//[^a-zA-Z0-9]/_}"

The next line removes the underscore prefix, tidying up the file name, however it isn't as intended.
I've just replaced this line with these:
> SAFENAME="${SAFENAME/#_/}" # Remove prefix
> SAFENAME="${SAFENAME/%_/}" # Remove suffix

Reply:
Aha, yeah, the second line didn't look right ;)